x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/profiler/component-template/profiling-hosts.json

```diff
@@ -0,0 +1,306 @@
+{
+  "template": {
+    "settings": {
+      "index": {
+        "number_of_replicas": 0,
+        "auto_expand_replicas": "0-1",
+        "refresh_interval": "10s",
+        "sort": {
+          "field": [
+            "profiling.project.id",
+            "host.id",
+            "@timestamp"
+          ]
+        }
+      },
+      "codec": "best_compression"
+    },
+    "mappings": {
+      "_source": {
+        "enabled": true
+      },
+      "properties": {
+        "ecs.version": {
+          "type": "keyword"
+        },
+        "host.id": {
+          "type": "keyword"
+        },
+        "@timestamp": {
+          "type": "date",
+          "format": "epoch_second"
+        },
+        "profiling": {
+          "dynamic": false,
+          "properties": {
+            "project.id": {
+              "type": "keyword"
+            },
+            "agent": {
+              "properties": {
+                "version": {
+                  "type": "version"
```

Comment:
The tests were failing because the `version` type is provided by a bundled plugin as opposed to being in core Elasticsearch. Tests don't include all bundled plugins by default and hence we need to enable this explicitly. I did this in https://github.com/elastic/elasticsearch/pull/96851/commits/6ea0926ebfca3ca1dc43a519406467334150ecf9.